src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -1855,8 +1866,34 @@ RESULT_CODE ProtocolHandlerImpl::HandleControlMessageRegisterSecondaryTransport(
   return RESULT_OK;
 }
 
+bool ProtocolHandlerImpl::ParseFullVersion(
+    utils::SemanticVersion& full_version,
+    const ProtocolFramePtr& packet) const {
+  SDL_LOG_AUTO_TRACE();
+
+  BsonObject request_params;
+  size_t request_params_size = bson_object_from_bytes_len(
+      &request_params, packet->data(), packet->total_data_bytes());
+  if (request_params_size > 0) {
+    char* version_param =
+        bson_object_get_string(&request_params, strings::protocol_version);
+    std::string version_string(version_param == NULL ? "" : version_param);
+    full_version = version_string;
+
+    // Constructed payloads added in Protocol v5
+    if (full_version.major_version_ < PROTOCOL_VERSION_5) {
+      return false;
+    }
+    bson_object_deinitialize(&request_params);
+  } else {
+    SDL_LOG_WARN("Failed to parse start service packet for version string");
```

Comment:
@VladSemenyuk should be return false here too?

Reply:
No. Bson data not mandatory.